hip/README.md

```diff
@@ -0,0 +1,130 @@
+Administration) responsible for the planning and preparation of a
+[capable exascale ecosystem](https://exascaleproject.org/what-is-exascale),
+including software, applications, hardware, advanced system engineering and early
+testbed platforms, in support of the nation’s exascale computing imperative.
+
+## Differences with the official benchmark version
+
+The HIP version differs from the official benchmark version of Laghos (in the
+top-level directory) in the following ways:
+
+1. Only problems 0 and 1 are defined
+2. Final iterations (`step`), time steps (`dt`) and energies (`|e|`) differ from the original version
+
+## Building
+
+Follow the steps below to build the HIP version with GPU acceleration.
+
+### Environment setup
+```sh
+export MPI_HOME=~/usr/local/openmpi/3.0.0
+```
+
+### Hypre
+- <https://computation.llnl.gov/projects/hypre-scalable-linear-solvers-multigrid-methods/download/hypre-2.11.2.tar.gz>
+- `tar xzvf hypre-2.11.2.tar.gz`
+- ` cd hypre-2.11.2/src`
+- `./configure --disable-fortran --with-MPI --with-MPI-include=$MPI_HOME/include --with-MPI-lib-dirs=$MPI_HOME/lib`
+- `make -j`
+- `cd ../..`
+
+### Metis
+-   <http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-5.1.0.tar.gz>
+-   `tar xzvf metis-5.1.0.tar.gz`
+-   `cd metis-5.1.0`
+-   ``make config prefix=`pwd` ``
+-   `make && make install`
+-   `cd ..`
+
+### MFEM
+-   `git clone git@github.com:mfem/mfem.git`
+-   `cd mfem`
+-   `git checkout laghos-v2.0`
+-   ``make config MFEM_USE_MPI=YES HYPRE_DIR=`pwd`/../hypre-2.11.2/src/hypre MFEM_USE_METIS_5=YES METIS_DIR=`pwd`/../metis-5.1.0``
+-   `make status` to verify that all the include paths are correct
+-   `make -j`
+-   `cd ..`
+
+### HIP Laghos
+-   `git clone git@github.com:CEED/Laghos.git`
+-   `cd Laghos/cuda`
+-   edit the `makefile`, set HIP\_ARCH to the desired architecture and the absolute paths to HIP\_DIR, MFEM\_DIR, MPI\_HOME
+-   `make` to build the HIP version
+
+## Running
+
+The HIP version can run the same sample test runs as the official benchmark
+version of Laghos.
+
+### Options
+-   -m <string>: Mesh file to use
+-   -ok <int>: Order (degree) of the kinematic finite element space
+-   -rs <int>: Number of times to refine the mesh uniformly in serial
+-   -p <int>: Problem setup to use, Sedov problem is '1'
+-   -cfl <double>: CFL-condition number
+-   -ms <int>: Maximum number of steps (negative means no restriction)
+-   -aware: Enable or disable MPI HIP Aware
+
+## Verification of Results
+
+To make sure the results are correct, we tabulate reference final iterations
+(`step`), time steps (`dt`) and energies (`|e|`) for the runs listed below:
+
+1. `mpirun -np 4 laghos -p 0 -m ../data/square01_quad.mesh -rs 3 -tf 0.75 -pa`
+2. `mpirun -np 4 laghos -p 0 -m ../data/cube01_hex.mesh -rs 1 -tf 0.75 -pa`
+3. `mpirun -np 4 laghos -p 1 -m ../data/square01_quad.mesh -rs 3 -tf 0.8 -pa -cfl 0.05`
+4. `mpirun -np 4 laghos -p 1 -m ../data/cube01_hex.mesh -rs 2 -tf 0.6 -pa -cfl 0.08`
```

Comment:
Why do you change the CFL?

Reply:
Same thing, it is due to the first `cuda` subdir, where I changed these values.
Now that the MFEM 4.0 port is ready, we should be able to get the same results.